### PR TITLE
TraitInfoInterface base for interfaces implemented by trait infos

### PR DESCRIPTION
**Pull request by atlimit8**
Neither @pchote nor myself liked using `ITraitInfo` for both trait infos and interfaces (for queries) implemented by trait infos.

### CODE REVIEW

File: OpenRA.Game/Traits/TraitsInterfaces.cs [hunk 1]
```diff
@@ -294,12 +294,13 @@ public interface ICrushable
 		bool CrushableBy(HashSet<string> crushClasses, Player owner);
 	}
 
-	public interface ITraitInfo { object Create(ActorInitializer init); }
+	public interface ITraitInfoInterface { }
+	public interface ITraitInfo : ITraitInfoInterface { object Create(ActorInitializer init); }
```
Comment (penev92) [inline OpenRA.Game/Traits/TraitsInterfaces.cs]: This inheritance here seems to defeat the whole purpose of this excercise.

**Comment by pchote** [comment]
Looks good to me, thanks :+1:

**Comment by Phrohdoh** [comment]
I'm with @penev92 and would like to see why / the usage for this.

**Comment by pchote** [comment]
It clearly distinguishes ITraitInfo (i.e. the trait info implementations) vs the IFooInfo interfaces that they can implement while not breaking everything involved with type constraints.

**Comment by Phrohdoh** [comment]
:+1: since this is for clearing up intent of code.

**Pull request merged**